File: articles/calculus-for-systems-modeling/articles/parameter-sweeps-and-sensitivity-analysis/fortran/parameter_sweeps_sensitivity.f90
program parameter_sweeps_sensitivity
  implicit none
  integer :: i, j
  real(8), dimension(5) :: rates
  real(8), dimension(4) :: caps
  real(8) :: value

  rates = (/0.18d0, 0.25d0, 0.35d0, 0.45d0, 0.55d0/)
  caps = (/80.0d0, 100.0d0, 125.0d0, 150.0d0/)

  print '(A)', 'growth_rate carrying_capacity initial_value stop_time final_value'
  do i = 1, 5
    do j = 1, 4
      value = logistic(20.0d0, 10.0d0, rates(i), caps(j))
      print '(5F16.8)', rates(i), caps(j), 10.0d0, 20.0d0, value
    end do
  end do

contains
  real(8) function logistic(t, x0, r, k)
    real(8), intent(in) :: t, x0, r, k
    logistic = k / (1.0d0 + ((k - x0) / x0) * exp(-r * t))
  end function
end program

File: articles/calculus-for-systems-modeling/articles/parameter-sweeps-and-sensitivity-analysis/c/parameter_sweeps_sensitivity.c
#include <math.h>
#include <stdio.h>
#include <stdlib.h>

static double logistic(double t, double x0, double r, double k){
  return k / (1.0 + ((k - x0) / x0) * exp(-r * t));
}
int main(void){
  const double rates[5] = {0.18, 0.25, 0.35, 0.45, 0.55};
  const double caps[4] = {80.0, 100.0, 125.0, 150.0};
  printf("growth_rate,carrying_capacity,initial_value,stop_time,final_value,output_metric,warning\n");
  for(int i=0; i<5; i++){
    for(int j=0; j<4; j++){
      double v = logistic(20.0, 10.0, rates[i], caps[j]);
      printf("%.6f,%.6f,10.000000,20.000000,%.12f,final_state_value,Sweep results depend on tested ranges baseline assumptions and model structure.\n", rates[i], caps[j], v);
    }
  }
  return EXIT_SUCCESS;
}
